### PR TITLE
CORTX-27894 - Use noarch architecture for HA package.

### PR DESCRIPTION
**Pull request by shailesh-vaidya**
# Problem Statement
-  cortx-ha package generation is failing in Rocky Linux 8.4. [build.sh](https://github.com/Seagate/cortx-ha/blob/main/jenkins/build.sh#L77) uses `x86_64` architecture for cortx-ha package. As cortx-ha contains Python scripts this should be arch indepndent. Removed `--force-arch x86_64` option from `python3.6 setup.py bdist_rpm bdist_rpm` command

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
-  [x] Coding conventions are followed and code is consistent

# Testing 
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM

# Review Checklist 
- [x] PR is self reviewed
- [x] JIRA number/GitHub Issue added to PR
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained
- [ ] Is there a change in filename/package/module or signature? [Y/N]: 
- [ ] If yes for above point, is a notification sent to all other cortx components? [Y/N]
- [x] Side effects on other features (deployment/upgrade)? [Y/N]
- [ ] Dependencies on other component(s)? [Y/N]
-     If yes for above point, post link to the corresponding PR.

# Review Checklist 
- [ ] Is perfline test run and the report with and without the changes updated in the PR? [Y/N]: 

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Review by satish-seagate — APPROVED** [review]

**Review by ujjwalpl — APPROVED** [review]

**Comment by ArchanaLimaye** [comment]
1N K8S deployment is failing. Can you please resolve that

**Comment by ArchanaLimaye** [comment]
Codacy is also not complete yet,

File: jenkins/build.sh [hunk 1]
```diff
@@ -73,15 +73,15 @@ rm -rf ${BASE_DIR}/dist
 cp -rf ${BASE_DIR}/conf/etc/v2/ha.conf.template ${BASE_DIR}/conf/etc/v2/ha.conf
 sed -i -e "s|<VERSION>|${version}|g" ${BASE_DIR}/conf/etc/v2/ha.conf
 
-python3.6 setup.py bdist_rpm --version="${version}" --install-dir="${HA_INSTALL_PATH}" --release="$BUILD" \
---force-arch x86_64 --post-install jenkins/rpm/v2/k8s/post_install_script.sh \
+python3.6 setup.py bdist_rpm --version="${version}" --install-dir="${HA_INSTALL_PATH}" --binary-only --release="$BUILD" \
```
Comment (ArchanaLimaye) [inline jenkins/build.sh:76]: binary-only flag will generate only binary and not source. Is that intended ? Will source files be available inside deployed container after this change ?

Comment (satish-seagate) [reply jenkins/build.sh:76]: >  Will source files be available inside the deployed container after this change?
Yes, I am able to see the source file inside the container after deployment.

Comment (satish-seagate) [reply jenkins/build.sh:76]: @ArchanaLimaye

Comment (satish-seagate) [reply jenkins/build.sh:76]: @shailesh-vaidya, Please confirm this comment.

**Review by indrajitzagade — APPROVED** [review]

**Review by mukhtar-inamdar — APPROVED** [review]

**Comment by ArchanaLimaye** [comment]
The 1NK8s deployment has failed and codacy is stuck.
@shailesh-vaidya can you please take a look at that as well

File: jenkins/build.sh (hunk 1 above)
Comment (shailesh-vaidya) [reply jenkins/build.sh:76]: @ArchanaLimaye @satish-seagate  - We do not need source rpm. ie. .src.rpm file. We were not copying it to dist directory also. refer - https://github.com/Seagate/cortx-ha/blob/main/jenkins/build.sh#L82. Hence disabled it's generation. 
Generated rpm has all the source files.

**Comment by shailesh-vaidya** [comment]
> The 1NK8s deployment has failed and codacy is stuck. @shailesh-vaidya can you please take a look at that as well

@mukul-seagate11 Could you please check for Codacy stuck issue. 
@ArchanaLimaye 1N deployment seems to e failing and it's looking for x86_64 based rpm and this patch does not have that. Let me check and confirm.

**Comment by shailesh-vaidya** [comment]
run vm-deployment

**Comment by shailesh-vaidya** [comment]
retest this please

**Comment by shailesh-vaidya** [comment]
> > The 1NK8s deployment has failed and codacy is stuck. @shailesh-vaidya can you please take a look at that as well
> 
> @mukul-seagate11 Could you please check for Codacy stuck issue. @ArchanaLimaye 1N deployment seems to be failing as it's looking for x86_64 based rpm and this patch does not have that. Let me check and confirm.

Thanks to @mukul-seagate11 Codacy is green now. 1N deployment also working fine now.

File: jenkins/build.sh (hunk 1 above)
Comment (ArchanaLimaye) [reply jenkins/build.sh:76]: okay

**Review by ArchanaLimaye — APPROVED** [review]

**Pull request merged**